src/BaGet.Core/Services/FileStorageService.cs

```diff
@@ -96,13 +99,19 @@ public Task DeleteAsync(string path, CancellationToken cancellationToken = defau
 
         private string GetFullPath(string path)
         {
-            // TODO: This should check that the result is in _storePath for security.
             if (string.IsNullOrEmpty(path))
             {
                 throw new ArgumentException("Path is required", nameof(path));
             }
 
-            return Path.Combine(_storePath, path);
+            // Verify path is under the _storePath.
+            string fullPath = Path.GetFullPath(Path.Combine(_storePath, path));
+            if (!fullPath.StartsWith(_storePath))
```

Comment:
If the `fullPath` and `_storePath` are identical, this check will pass. This causes the test `fullPath + Path.DirectorySeparatorChar;` to fail. I would add something like:

```
if (fullPath.Equals(_storePath, StringComparison.InvariantCultureIgnoreCase))
{
    throw new ArgumentException("Path must be a file within the store path", nameof(path));
}
```

Reply:
I've added a length check. I also changed the startswith compare to an Ordinal one.